python/ql/src/semmle/python/security/strings/Basic.qll

```diff
@@ -73,9 +73,6 @@ private predicate str_format(ControlFlowNode fromnode, CallNode tonode) {
     tonode.getFunction().(AttrNode).getName() = "format" and
     (
         tonode.getAnArg() = fromnode
-        or
-        // TODO: if this case is not covered by tonode.getAnArg(), we should change it so it is :\
-        tonode.getNode().getAKeyword().getValue() = fromnode.getNode()
```

Comment:
I'm really puzzling over what exactly this was supposed to do. Can you add a test case that exhibits the bad flow that this would cause?

Reply:
I only have a very informal idea about how splitting works, but this is the scenario I'm worried about

```py
if is_not_tainted(x): # splitting happens here
    ...

y = "Some formatting {x}!".format(x=x)

if is_not_tainted(x):
    dangerous(y)
```

So the idea is that we have 2 control-flow-nodes for the call to format, one where x is tainted, and one where x is not. However, if the code I deleted was still active, we would end up saying there is a taint-step from the `x` that is tainted, to the format call in the splitted control-flow-graph where `x` was not tainted, resulting in `y` being tainted in **both** paths.

> I'm really puzzling over what exactly this was supposed to do.

I think it was a mistake :neutral_face:

Reply:
Okay, yeah. That issue with splitting makes sense.
Had a quick look at `git blame`, and it seems this was present from the [very beginning](https://git.semmle.com/Semmle/code/commit/e1f94b696f591b1546e910f7db65635a670e3ed6#diff-4fbf4dd04c327e072cb9e420082f5c25R140).

Reply:
Does that mean you're happy about this PR? :wink: :wink:

Reply:
_Almost_. I think we can dispense with the test, but don't we have a set of superfluous parentheses now? :slightly_smiling_face:

Reply:
dang :smile: